platforms/allwinner-d1/core/src/lib.rs

```diff
@@ -234,8 +234,10 @@ impl D1 {
     /// At the moment, we only service the Channel 0 interrupt,
     /// which indicates that the serial transmission is complete.
     fn handle_dmac() {
+        let _isr = kernel::isr::Isr::enter();
```

Comment:
We could potentially do this at the top of the actual ISR entry point instead of in every handler

Reply:
done!